### PR TITLE
add custom monitoring guide

### PR DESCRIPTION
**Pull request by huanggze**
Signed-off-by: huanggze <loganhuang@yunify.com>

### CODE REVIEW

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+![set-servicemonitor-to-true](/images/docs/project-user-guide/custom-application-monitoring/set-servicemonitor-to-true.PNG)
+
+{{< notice warning >}}
+Don't forget to enable SericeMonitor CRD if you are using external exporter helm charts. Those charts usually disable ServiceMonitor by default and require manual modification.
+{{</ notice >}}
+
+- Modify MySQL connection parameters
+
+MySQL exporter needs to connect to the target MySQL. In this tutorial, MySQL is installed with the Service name `mysql-egn701`. So we modify `mysql.host` to `mysql-egn701`, `mysql.pass` to `testing`, and `user` to `root`. Note that your MySQL Service may be created with a different name!
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md:73]: So we modify `mysql.host` to `mysql-egn701`, `mysql.pass` to `testing`, and `user` to `root`. Note that your MySQL Service may be created with a different name!
->
Modify `mysql.host` to `mysql-egn701`, `mysql.pass` to `testing`, and `user` to `root` as below. Note that your MySQL Service may be created with **a different name**.

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+- Click **Deploy** and wait until MySQL exporter is up
+
+![exporter-click-deploy-2](/images/docs/project-user-guide/custom-application-monitoring/exporter-click-deploy-2.PNG)
+
+![exporter-is-running](/images/docs/project-user-guide/custom-application-monitoring/exporter-is-running.PNG)
+
+### Step 3: Create Dashboard
+
+After two minutes, you can create monitoring dashboard for MySQL and visualize metrics in real time.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md:85]: After two minutes, you can create monitoring dashboard for MySQL and visualize metrics in real time.
->
After about two minutes, you can create a monitoring dashboard for MySQL and visualize metrics in real time.

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+---
+
+This section walks you through monitoring a sample web application. The application is instrumented with Prometheus Golang client in its code. Therefore, it can expose metrics directly without the help of exporters.
+
+## Prerequisites
+
+- Please make sure you enable the OpenPitrix system.
+- You need to create a workspace, a project, and a user account for this tutorial. This account needs to be a platform regular user and to be invited as a self provisioner. In this tutorial, we will work with the workspace `my-workspace` and the project `demo`. We will log in as `project-self-provisioner` and create the project `demo` beforehand.
+- You know Helm Chart.
+- You know [PromQL](https://prometheus.io/docs/prometheus/latest/querying/examples/).
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md:17]: ->
- Knowledge of Helm Chart and [PromQL](https://prometheus.io/docs/prometheus/latest/querying/examples/).

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+## Hands-on Lab
+
+### Step 1: Prepare Sample Web Application Image
+
+First, prepare the sample web application image, The sample web application exposes a user-defined metric called `myapp_processed_ops_total`. It is a counter type metric that counts the number of operations that have been processed by far. The counter auto increases by one every 2 seconds.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md:23]: prepare the sample web application image,
->
prepare the sample web application image.

File: content/en/docs/project-user-guide/custom-application-monitoring/visualization/overview.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+linkTitle: "Overview"
+weight: 2130
+---
+
+This section introduces dashboard features. You will learn how to visualize metric data on KubeSphere for your custom applications. If you do not know how to integrate your application metrics into KubeSphere monitoring system, you should read [Introduction](../../introduction) first.
+
+## Create Dashboard
+
+To create new dashboards for your application metrics, you should navigate to **Custom Monitoring** on the Project Overview page. There are free ways to create dashboards. For MySQL, Elasticsearch, and Redis, you can use built-in templates. These templates are for demonstration purposes and are updated with KubeSphere releases. Besides, you can choose to customize dashboards from scratch.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/custom-application-monitoring/visualization/overview.md:14]: There are free ways to create dashboards.
->
There are different ways to create dashboards.

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+linkTitle: "Monitor Sample Web"
+weight: 2120
+---
+
+This section walks you through monitoring a sample web application. The application is instrumented with Prometheus Golang client in its code. Therefore, it can expose metrics directly without the help of exporters.
+
+## Prerequisites
+
+- Please make sure you enable the OpenPitrix system.
+- You need to create a workspace, a project, and a user account for this tutorial. This account needs to be a platform regular user and to be invited as a self provisioner. In this tutorial, we will work with the workspace `my-workspace` and the project `demo`. We will log in as `project-self-provisioner` and create the project `demo` beforehand.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md:15]: At project level, there are only three roles: viewer, operator and admin. Just want to make sure here: is the account `project-self-provisioner` used here assigned with the role `workspace-self-provisioner`?

**Review by Sherlock113 — COMMENTED** [review]
Please have a look at my comments and make necessary changes. Thanks.

Others look good to me in general. Minor wording issues exist but won't affect understanding. I will update them later after this pr is merged.

File: content/en/docs/project-user-guide/custom-application-monitoring/introduction.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+## Overview
+
+The kubeSphere monitoring engine is powered by Prometheus and Prometheus Operator. To integrate custom application metrics into KubeSphere, you need to go through the following steps in general.
+
+### Expose Prometheus-Formatted Metrics
+
+First of all, your application must expose Prometheus-formatted metrics. Prometheus exposition format is the de-facto format in the realm of cloud-native monitoring. Prometheus uses a [text-based exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/). Depending on your application and use case, there are two ways to expose metrics:
+
+#### direct exposing
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/introduction.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/introduction.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+Directly exposing Prometheus metrics from applications is a common way among cloud-native applications. It requires developers to import Prometheus client libraries in their codes and expose metrics at a specific endpoint. Many applications, such as ETCD, CoreDNS, and Istio, adopt this method.
+
+Prometheus community offers client libraries for most programming languages. Please find your language on [Prometheus Client Libraries](https://prometheus.io/docs/instrumenting/clientlibs/) page. For Golang developer, read [Instrumenting a Go application](https://prometheus.io/docs/guides/go-application/) to learn how to write a Prometheus-compliant application.
+
+The [sample web application](../get-started/monitor-sample-web) is an example demonstrating how an application exposes Prometheus-formatted metrics directly. 
+
+#### indirect exposing
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/introduction.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/introduction.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+linkTitle: "Introduction"
+weight: 2100
+---
+
+Custom monitoring allows you to monitor and visualize custom application metrics on KubeSphere. The application can be either a third-party application, such as MySQL, Redis, and Elasticsearch, or your web application. This section introduces the workflow of this feature.
+
+## Overview
+
+The kubeSphere monitoring engine is powered by Prometheus and Prometheus Operator. To integrate custom application metrics into KubeSphere, you need to go through the following steps in general.
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/introduction.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/introduction.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+### Expose Prometheus-Formatted Metrics
+
+First of all, your application must expose Prometheus-formatted metrics. Prometheus exposition format is the de-facto format in the realm of cloud-native monitoring. Prometheus uses a [text-based exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/). Depending on your application and use case, there are two ways to expose metrics:
+
+#### direct exposing
+
+Directly exposing Prometheus metrics from applications is a common way among cloud-native applications. It requires developers to import Prometheus client libraries in their codes and expose metrics at a specific endpoint. Many applications, such as ETCD, CoreDNS, and Istio, adopt this method.
+
+Prometheus community offers client libraries for most programming languages. Please find your language on [Prometheus Client Libraries](https://prometheus.io/docs/instrumenting/clientlibs/) page. For Golang developer, read [Instrumenting a Go application](https://prometheus.io/docs/guides/go-application/) to learn how to write a Prometheus-compliant application.
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/introduction.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+linkTitle: "Monitor MySQL"
+weight: 2110
+---
+From the [Introduction](../../introduction#indirect-exposing) section, you know it is not feasible to instrument MySQL with Prometheus metrics directly. To expose MySQL metrics in Prometheus format, you need to deploy MySQL exporter instead.
+
+This tutorial walks you through an example of how to monitor MySQL metrics and visualize them.
+
+## Prerequisites
+
+- Please make sure you enable the OpenPitrix system. We will deploy MySQL and MySQL exporter from the App Store.
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+- You need to create a workspace, a project, and a user account for this tutorial. The account needs to be a platform regular user and to be invited as the project operator. In this tutorial, we will work with the workspace my-workspace, and the project demo. We will log in as `project-operator`.
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+
+![go-to-app-store](/images/docs/project-user-guide/custom-application-monitoring/go-to-app-store.PNG)
+
+- Find **MySQL** and Click **Deploy**
+
+![find-mysql](/images/docs/project-user-guide/custom-application-monitoring/find-mysql.PNG)
+
+![click-deploy](/images/docs/project-user-guide/custom-application-monitoring/click-deploy.PNG)
+
+- Make sure MySQL to be deployed in `demo`
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+![uncomment-mysqlRootPassword](/images/docs/project-user-guide/custom-application-monitoring/uncomment-mysqlRootPassword.PNG)
+
+- Wait until MySQL is up
+
+![check-if-mysql-is-running](/images/docs/project-user-guide/custom-application-monitoring/check-if-mysql-is-running.PNG)
+
+
+### Step 2: Deploy MySQL exporter
+
+Deploy MySQL exporter in `demo` on the same cluster. MySQL export is responsible for querying MySQL status and reports them in Prometheus format.
```
[the remark this answers is not in the document]
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md (hunk 1 above)
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md (hunk 2 above)
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-mysql.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md (hunk 3 above)
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md (hunk 4 above)
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md]: done

File: content/en/docs/project-user-guide/custom-application-monitoring/visualization/overview.md (hunk 5 above)
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/visualization/overview.md]: free -> three

File: content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md (hunk 6 above)
Comment (huanggze) [reply content/en/docs/project-user-guide/custom-application-monitoring/get-started/monitor-sample-web.md]: yes, it is. I paraphrased and clarified this part

**Comment by Sherlock113** [comment]
just one thing to make it clearer here. Confirmed with @huanggze , this should be workspace-self-provisioner, but the image in the guide is project-self-provisioner.  

截屏2020-09-29下午2 51 46

@FeynmanZhou You can merge this pr now. I will update images later together with some minor changes. Others look good to me. Thanks.

**Pull request merged**